modules/micromodeconnection/index.js

```diff
@@ -0,0 +1,260 @@
+        }
+    };
+
+    sendMessageToMicroWindow('mainWindowPeerCreated', null);
+}
+
+/**
+ * Creates peer connection offer to micro window.
+ */
+function createAndSendOffer() {
+    log("main window createOffer start");
+
+    const offerOptions = {
+        offerToReceiveAudio: 1,
+        offerToReceiveVideo: 1
+    };
+    peerConnectionMain.createOffer(
+        offerOptions
+    ).then(
+        onCreateOfferSuccess,
+        onCreateOfferError
+    );
+}
+function onCreateOfferError(error) {
+    log("main window failed to create session description: "+error.toString());
+}
+
+/**
+ * Sends peer connection offer created to micro window.
+ */
+function onCreateOfferSuccess(desc) {
+    log("offer from main window\n" + desc.sdp);
+    log("main window setLocalDescription start");
+    peerConnectionMain.setLocalDescription(desc).then(
+        function() {
+            log("main window setLocalDescription complete");
+        },
+        function(error) {
+            log("failed to set session description: "+error.toString());
+        }
+    );
+
+    const data = {type: desc.type, sdp: desc.sdp};
+    sendMessageToMicroWindow('mainWindowLocalDescriptionSet', data);
+}
+
+/**
+ * Sends message from main window to mircro window.
+ */
+function sendMessageToMicroWindow(message, data) {
+  args = [];
+  args[0] = message;
+  args[1] = data;
+    ipcRenderer.send('mainWindowEvent', args);
+}
+
+
+/*------------------------------Micro Window----------------------------------*/
+let peerConnectionMicro;
+let remoteVideo;
+
+/**
+ * Initiates ipc listners for messages from the main window.
+ * @param {object} video - video object on the micro window.
+ */
+function initMicroConnection(video) {
+    remoteVideo = video;
+    ipcRenderer.on('mainWindowPeerCreated', () => {
+        setupMicroConnection();
+    });
+    ipcRenderer.on('mainWindowLocalDescriptionSet', (event, desc) => {
+        createAnswer(desc);
+    });
+}
+
+/**
+ * Creates micro window's RTCPeerConnection object and set ice candidate listneres.
+ */
+function setupMicroConnection() {
+    peerConnectionMicro = new RTCPeerConnection();
+    log("created micro window peer connection object");
+
+    ipcRenderer.on('mainWindowIceCandidate', (event, candidate) => {
+        peerConnectionMicro.addIceCandidate(candidate)
+        .then(
+            function() {
+                log('micro window addIceCandidate success');
+            },
+            function(err) {
+                log('micro window failed to add ICE Candidate: '+err.toString());
+            });
+    });
+
+    peerConnectionMicro.onicecandidate = function(event) {
+        if(event.candidate !== null) {
+            const newIceCandidate = {
+                candidate: event.candidate.candidate,
+                sdpMLineIndex: event.candidate.sdpMLineIndex,
+                sdpMid: event.candidate.sdpMid
+            };
+            sendMessageToMainWindow('microWindowIceCandidate', newIceCandidate);
```

Comment:
why not send the candidate just as it comes? it would also help with signaling the end of candidates.